### PR TITLE
Resize Linode

### LINKED ISSUE #1193: RESIZE hookup to API

### PR DESCRIPTION
**Pull request by na3d**
Closes #1193

### CODE REVIEW

File: test/linodes/linode/layouts/ResizePage.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    await page.instance().onSubmit();
+
+    const fn = dispatch.firstCall.args[0];
+    await expectRequest(fn, '/linode/instances/1234/resize', () => {}, null,
+      d => {
+        expect(d.method).to.equal('POST');
+        expectObjectDeepEquals(JSON.parse(d.body), {
+          type: 'linode1024.5',
+        });
+      }
```
Comment (eatonphil) [inline test/linodes/linode/layouts/ResizePage.spec.js]: Like last time, make this a bit easier on yourself:

```
await expectRequest(fn, '/linode/instances/1234/resize', undefined, undefined, {
  method: "POST",
  body: {
    type: 'linode1024.5',
  }
});
```

Comment (na3d) [inline test/linodes/linode/layouts/ResizePage.spec.js]: Oh yeah, just breaking bad habits.

Comment (na3d) [inline test/linodes/linode/layouts/ResizePage.spec.js]: Thank you, nice catch.

**Comment by mparke** [comment]
a few questions:

- do we have an issue to fix the display of plans?

- do we have an issue to look at disabling actions if we know a linode is busy? this falls into the realm of faster polling on object/ endpoints to get status after things like boot/reboot/resize

- can we create issues for revisiting messaging (needs to be communicated to api)
screen shot 2017-03-02 at 10 55 18 am
screen shot 2017-03-02 at 10 57 48 am

**Comment by eatonphil** [comment]
> do we have an issue to look at disabling actions if we know a linode is busy? this falls into the realm of faster polling on object/ endpoints to get status after things like boot/reboot/resize

The API doesn't (and perhaps will not) supply this information to us, unfortunately.

**Comment by na3d** [comment]
> do we have an issue to fix the display of plans?

Yeah 1299 will fix this for anywhere plans are displayed.

**Comment by abemassry** [comment]
This does work, but I'm going to make a separate issue to have the page either navigate somewhere else or show some indicator upon success, because it wasn't apparent to me when it actually worked cause it "just worked":tm:

**Pull request merged**